akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -3026,6 +3026,32 @@ trait FlowOps[+Out, +Mat] {
   def concat[U >: Out, Mat2](that: Graph[SourceShape[U], Mat2]): Repr[U] =
     internalConcat(that, detached = true)
 
+  /**
+   * Concatenate the given [[Source]]s to this [[Flow]], meaning that once this
+   * Flow’s input is exhausted and all result elements have been generated,
+   * the [[Source]]s' elements will be produced.
+   *
+   * Note that the [[Source]]s are materialized together with this Flow and is "detached" meaning it will
+   * in effect behave as a one element buffer in front of both the sources, that eagerly demands an element on start
+   * (so it can not be combined with `Source.lazy` to defer materialization of `that`).
+   *
+   * The second source is then kept from producing elements by asserting back-pressure until its time comes.
+   *
+   * When needing a concat operator that is not detached use [[#concatLazy]]
+   *
+   * If this [[Flow]] gets upstream error - no elements from the given [[Source]]s will be pulled.
+   *
+   * '''Emits when''' element is available from current stream or from the given [[Source]]s when current is completed
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' given all those [[Source]]s completes
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def concatAll[U >: Out](those: Graph[SourceShape[U], _]*): Repr[U] =
+    internalConcatAll(those.toList, detached = true)
```

Comment:
I wonder if it is too crazy to only have one `concatAll` which is the lazy one, and skip the detached overload completely. Mismatch with `concat`/`concatLazy` but maybe worth it since the detachedness there is what is weird?

Reply:
I think so too, but the name...